jit/src/lib.rs

```diff
@@ -143,3 +211,54 @@ impl fmt::Debug for CompiledCode {
         f.write_str("[compiled code]")
     }
 }
+
+pub struct ArgsBuilder<'a> {
+    values: Vec<MaybeUninit<UnTypedAbiValue>>,
+    initialised: Vec<bool>,
+    code: &'a CompiledCode,
+}
+
+impl<'a> ArgsBuilder<'a> {
+    fn new(code: &'a CompiledCode) -> ArgsBuilder<'a> {
+        ArgsBuilder {
+            values: vec![MaybeUninit::uninit(); code.sig.args.len()],
```

Comment:
Couldn't this be `Vec<Option<UntypedAbiArg>>`? and you could like do `self.values.iter().map(|val| val.as_ref().map(|val| middle::Abi::new(val)).collect::<Option<_>>()`

And you could just store the Vec<Option<>> in the Args struct, since it just matters that it's not dropped